### PR TITLE
Bluetooth: Host: Add various info/err logging

### PR DESCRIPTION
**Pull request by alwa-nordic**
Make use of `BT_INFO` and `BT_WARN` to log big events related to connections.

### CODE REVIEW

File: subsys/bluetooth/host/conn.c [hunk 1]
```diff
@@ -1149,6 +1149,10 @@ void bt_conn_unref(struct bt_conn *conn)
 
 	__ASSERT(old > 0, "Conn reference counter is 0");
 
+	if (old == 1) {
+		BT_INFO("conn %d recycled", bt_conn_index(conn));
```
Comment (jhedberg) [inline subsys/bluetooth/host/conn.c:1153]: Should he BT_DBG()

File: subsys/bluetooth/host/conn.c [hunk 2]
```diff
@@ -1336,6 +1340,14 @@ static void notify_connected(struct bt_conn *conn)
 {
 	struct bt_conn_cb *cb;
 
+	if (conn->err) {
+		BT_WARN("connect failed, conn %d, dst %s, error %d", bt_conn_index(conn),
+			bt_addr_le_str(bt_conn_get_dst(conn)), conn->err);
+	} else {
+		BT_INFO("connected, conn %d, dst %s", bt_conn_index(conn),
```
Comment (jhedberg) [inline subsys/bluetooth/host/conn.c:1347]: Also, BT_DBG()

File: subsys/bluetooth/host/conn.c [hunk 3]
```diff
@@ -1353,6 +1365,8 @@ static void notify_disconnected(struct bt_conn *conn)
 {
 	struct bt_conn_cb *cb;
 
+	BT_INFO("disconnected, conn %d", bt_conn_index(conn));
```
Comment (jhedberg) [inline subsys/bluetooth/host/conn.c:1368]: Same here

File: subsys/bluetooth/host/smp.c [hunk 4]
```diff
@@ -1906,6 +1906,8 @@ static void smp_pairing_complete(struct bt_smp *smp, uint8_t status)
 			bt_keys_store(conn->le.keys);
 		}
 
+		BT_INFO("pairing_complete, conn %d, bonded %d", bt_conn_index(conn), bond_flag);
```
Comment (jhedberg) [inline subsys/bluetooth/host/smp.c:1909]: And here

File: subsys/bluetooth/host/smp.c [hunk 5]
```diff
@@ -3936,7 +3941,7 @@ static uint8_t smp_ident_addr_info(struct bt_smp *smp, struct net_buf *buf)
 	struct bt_smp_ident_addr_info *req = (void *)buf->data;
 	uint8_t err;
 
-	BT_DBG("identity %s", bt_addr_le_str(&req->addr));
+	BT_INFO("identity, conn %d, %s", bt_conn_index(conn), bt_addr_le_str(&req->addr));
```
Comment (jhedberg) [inline subsys/bluetooth/host/smp.c:3944]: And here

**Review by jhedberg — CHANGES_REQUESTED** [review]
BT_INFO() should be used only in very rare cases, since it can’t be turned off if you have debug logs enabled. The only places we use it (IIRC) are for one time generally useful information when initializing the stack.

**Comment by alwa-nordic** [comment]
@jhedberg

> BT_INFO() should be used only in very rare cases, since it can’t be turned off if you have debug logs enabled.

I assume you are saying that BT_INFO cannot be disabled while enabling logging from other modules. If that's true, then I think we have problem with the logging system.

I really want a log level that prints high-level info that correspond to real-world events, without spewing out bazillions of pages about what the implementation does internally. Maybe there should be a TRACE level that  is the most verbose one and DEBUG could be for high-level events. But I think INFO is underused. RFC @jori-nordic ?

The reason I added these in the first place is that I gathered pretty much every sample and Babelsim test prints out these events. So it's clearly not just me who thinks this information is useful.

**Comment by jori-nordic** [comment]
> @jhedberg
> 
> > BT_INFO() should be used only in very rare cases, since it can’t be turned off if you have debug logs enabled.
> 
> I assume you are saying that BT_INFO cannot be disabled while enabling logging from other modules. If that's true, then I think we have problem with the logging system.
> 
> I really want a log level that prints high-level info that correspond to real-world events, without spewing out bazillions of pages about what the implementation does internally. Maybe there should be a TRACE level that is the most verbose one and DEBUG could be for high-level events. But I think INFO is underused. RFC @jori-nordic ?
> 
> The reason I added these in the first place is that I gathered pretty much every sample and Babelsim test prints out these events. So it's clearly not just me who thinks this information is useful.

I think he means if you enable `BT_DEBUG_LOG`, you pretty much don't have control of the logging severity anymore, the INFO level is enabled unconditionally for every bluetooth module. the only thing you have control over is enabling the DBG level, e.g. with `BT_DEBUG_SMP`.

I think that sucks too (given that changing the logging severity works just fine at runtime, tested with the shell), and would like to see `BT_xx` logging functions deprecated.

**Comment by jhedberg** [comment]
@jori-nordic yes, that’s what I mean. And I’m all in favor of moving to standard logging macros instead of Bluetooth specific ones, if someone wants to pick up that task.

**Comment by alwa-nordic** [comment]
I'll move this PR into draft mode then. I think it will be more relevant after we address the move to standard logging macros.

**Review by Thalley — CHANGES_REQUESTED** [review]
Agree with Johan's comments

**Comment by jori-nordic** [comment]
@alwa-nordic any plans to continue with this?

**Comment by alwa-nordic** [comment]
> @alwa-nordic any plans to continue with this?

Have we made progress on disabling info level by default?

**Pull request closed without merging**